apps/prairielearn/src/tests/groupExam.test.ts

```diff
@@ -0,0 +1,513 @@
+import { assert } from 'chai';
+import cheerio = require('cheerio');
+import fetch from 'node-fetch';
+import { config } from '../lib/config';
+import { step } from 'mocha-steps';
+
+import { queryAsync, queryOneRowAsync, loadSqlEquiv } from '@prairielearn/postgres';
+const sql = loadSqlEquiv(__filename);
+
+import helperServer = require('./helperServer');
+import { TEST_COURSE_PATH } from '../lib/paths';
+
+const siteUrl = 'http://localhost:' + config.serverPort;
+const baseUrl = siteUrl + '/pl';
+const courseInstanceUrl = baseUrl + '/course_instance/1';
+
+const storedConfig: any = {};
+
+const GROUP_NAME = 'groupBB';
+const GROUP_NAME_ALTERNATIVE = 'groupCC';
+
+interface StudentUser {
+  user_id: string;
+  uid: string;
+  name: string;
+  uin: string;
+}
+
+async function generateThreeStudentUsers(): Promise<StudentUser[]> {
+  const result = await queryAsync(sql.generate_and_enroll_3_users, []);
+  assert.lengthOf(result.rows, 3);
+  return result.rows;
+}
+
+/**
+ * Switches active user and loads assessment. Asserts the number of CSRF tokens on the page
+ * after loading the given assessment URL.
+ */
+async function switchUserAndLoadAssessment(
+  studentUser: StudentUser,
+  assessmentUrl: string,
+  authUin: string,
```

Comment:
A `StudentUser` (first parameter here) already has a `uin` property - is there a reason for providing it separately here?

Reply:
Another pattern match from older group work testing code; the `authUin` was always provided manually in these tests, but it works using the `StudentUser`'s `uin` as well! The separate parameter isn't needed then